src/Native/MatrixFactorizationNative/UnmanagedMemory.cpp

```diff
@@ -115,7 +115,7 @@ EXPORT_API(float) MFCrossValidation(const mf_problem_bridge *prob_bridge, int32_
 {
     auto param = TranslateToParam(param_bridge);
     auto prob = TranslateToProblem(prob_bridge);
-    return mf_cross_validation(&prob, nr_folds, param);
+    return static_cast<float>(mf_cross_validation(&prob, nr_folds, param));
```

Comment:
It appears `mf_cross_validation` returns `mf_double`

[https://github.com/cjlin1/libmf/blob/dadfd21ef3f6af633bb1b4c35f79a0da8df5be82/mf.h#L98](https://github.com/cjlin1/libmf/blob/dadfd21ef3f6af633bb1b4c35f79a0da8df5be82/mf.h#L98)

which is a `double`

[https://github.com/cjlin1/libmf/blob/dadfd21ef3f6af633bb1b4c35f79a0da8df5be82/mf.h#L16](https://github.com/cjlin1/libmf/blob/dadfd21ef3f6af633bb1b4c35f79a0da8df5be82/mf.h#L16)

Is `static_cast` really the appropriate thing to do here?

Reply:
Would it be more appriopriate to return a `double` instead?

---
In reply to: [276750656](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276750656) [](ancestors = 276750656)

Reply:
Perhaps, but that would require changes all the way through to the C#  side.  Do we want to do this?  In most cases ML.NET uses single precision anyway...

---
In reply to: [276751151](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276751151) [](ancestors = 276751151,276750656)